File: 백준/Silver/2644. 촌수계산/촌수계산.py
n = int(input())    # 전체 사람의 수
p1, p2 = map(int, input().split())
m = int(input())    # 부모 자식들 간의 관계 수
relation = [[] for _ in range(n+1)]
visited = [False] * (n+1)
arrive = False

def dfs(s, cnt):
    global count, arrive
    visited[s] = True
    cnt += 1

    if s == p2:
        print(cnt-1)
        arrive = True
        return

    for j in relation[s]:
        if not visited[j]:
            dfs(j, cnt)

for i in range(m):
    parent, child = map(int, input().split())    # 부모의 자식 간의 관계

    relation[parent].append(child)
    relation[child].append(parent)

dfs(p1, 0)
if arrive == False:
    print(-1)

File: 백준/Silver/2644. 촌수계산/촌수계산.c
#include <stdio.h>

int rel[101][101];
int visited[101];
int ans = -1;

void dfs(start, end, cnt) {
	
	visited[start] = 1;

	if (start == end) {
		ans = cnt;
		return;
	}

	for (int i=0; i < 101; i++) {
		if (rel[start][i] && !visited[i]) {
			dfs(i, end, cnt+1);
		}
	}
}

int main(void) {
	
	int n;    // 전체 사람의 수
	scanf("%d", &n);

	int c1, c2;		// 촌수를 계산해야 하는 번호
	scanf("%d %d", &c1, &c2);

	int m;		// 부모 자식들 간의 관계 수
	scanf("%d", &m);

	for (int i=0; i < m; i++) {
		int x, y;
		scanf("%d %d", &x, &y);
		rel[x][y] = 1;
		rel[y][x] = 1;
	}

	dfs(c1, c2, 0);
	printf("%d\n", ans);

	return 0;
}